niceml/dlframeworks/tensorflow/callbacks/csvlogger.py

```diff
@@ -26,13 +26,37 @@ class CSVLogger(Callback):
     def __init__(
         self, experiment_context: ExperimentContext, filename: str = "train_logs.csv"
     ):
+        """Callback that streams epoch results to a CSV file.
+
+        Supports all values that can be represented as a string,
+        including 1D iterables such as np.ndarray.
+
+        # Example
+        ```python
+        csv_logger = CSVLogger(exp_context=exp_context)
+        model.fit(X_train, Y_train, callbacks=[csv_logger])
+        ```
+
+        # Arguments
+            exp_context: An instance of ExperimentContext which is used to write csv files
+            separator: string used to separate elements in the CSV file.
+        """
+
         self.filename = filename
         self.keys = None
         self.data = None
         self.experiment_context = experiment_context
         super().__init__()
 
     def on_epoch_end(self, epoch, logs=None):
+        """
+        The on_epoch_end function is called at the end of every epoch.
+        It flushes the logs to a CSV file.
```

Comment:
replace 'flushes' with 'deletes' (or a better word describing what is done)

Reply:
I replace "flushes" with "writes"